### ISSUE TITLE
Toast: Examples are not fully accessible

### ISSUE THREAD

**Issue by NicholasBoll** [Workday/canvas-kit#429]
### Summary
Accessibility raised an issue with our Toast examples. Developers have raised a similar issue. We have no example of how to use toast messages properly within an application context. Without this context, accessibility doesn't make sense and we have no specifications for applications to follow. We can either create a `ToastContainer` component or create a pattern. A contextual working example will help accessibility with specifications and help developers know what specifications are required for toast messages in their own applications.

Current Toast stories:
* https://workday.github.io/canvas-kit/?path=/story/components-popups-toast-react--error
* https://workday.github.io/canvas-kit/?path=/story/components-popups-toast-react--successful

### Original bug (from CANVAS-624)

The "Toast" success and error patterns that do not contain a focusable element are status messages.

These status messages lack the appropriate WAI-ARIA roles, states and properties to ensure that they are correctly exposed to assistive technologies like a screen reader.

Please refer to detailed implementation guidance in LRN-17107

**Polite regions**

A polite live region indicates that the screen reader should wait until the user is idle before presenting updates to the user. The green "tick" icon indicates this "toast" is a polite notification.

Polite regions have the following role states and properties.
 * role="status" – creates the status live region
 * aria-live="polite" – provides a fallback for older screen readers / browsers
 * aria-atomic="true" – causes the entire text inside the live region to be read rather than only the text that has changed

Code sample

```html
<div role="status" aria-live="polite" aria-atomic="true">
       <!-- live region content is injected here -->
 </div>
```

![CleanShot 2020-01-22 at 15 53 29](https://user-images.githubusercontent.com/338257/73282366-be3cec00-41ae-11ea-99d9-538b2b75b54b.png)

**Assertive live regions**

Assertive live region indicates that the screen reader should interrupt the user to present the update to the user. The red "exclamation" icon indicates this "toast" is an assertive notification.

Assertive live regions have the following roles states and properties
 * role="alert" – creates the status live region
 * aria-live="assertive" – provides a fallback for older screen readers / browsers
 * aria-atomic="true" – causes the entire text inside the live region to be read rather than only the text that has changed

Code sample

```html
<div role="alert" aria-live="assertive" aria-atomic="true">
       <!-- live region content is injected here -->
 </div>
```

![CleanShot 2020-01-22 at 16 08 23](https://user-images.githubusercontent.com/338257/73282378-c5fc9080-41ae-11ea-85d0-98b62fde1dc3.png)
